### PR TITLE
Refine runtime feature discovery python API and add documentation to …

### PR DESCRIPTION
**Pull request by larroy**
…Python API docs

## Description ##
see title

@aaronmarkham 

### Essentials ###
Please feel free to remove inapplicable items for your PR.
- [x] The PR title starts with [MXNET-$JIRA_ID], where $JIRA_ID refers to the relevant [JIRA issue](https://issues.apache.org/jira/projects/MXNET/issues) created (except PRs with tiny changes)
- [x] Changes are complete (i.e. I finished coding on this PR)
- [x] All changes have test coverage:
- Unit tests are added for small changes to verify correctness (e.g. adding a new operator)
- Nightly tests are added for complicated/long-running ones (e.g. changing distributed kvstore)
- Build tests will be added for build configuration changes (e.g. adding a new build option with NCCL)
- [x] Code is well-documented: 
- For user-facing API changes, API doc string has been updated. 
- For new C++ functions in header files, their functionalities and arguments are documented. 
- For new examples, README.md is added to explain the what the example does, the source of the dataset, expected performance on test set and reference to the original paper if applicable
- Check the API doc at http://mxnet-ci-doc.s3-accelerate.dualstack.amazonaws.com/PR-$PR_ID/$BUILD_ID/index.html
- [x] To the my best knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change

### Changes ###
- [x]  Decode feature name to string since it comes as a buffer
- [x]  Add __repr__ as beautifier

### CODE REVIEW

File: docs/api/python/libinfo/libinfo.md [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+ ✔ BLAS_APPLE,
+ ✔ LAPACK,
+ ✔ MKLDNN,
+ ✖ OPENCV,
+ ✔ CAFFE,
+ ✔ PROFILER,
+ ✔ DIST_KVSTORE,
+ ✔ CXX14,
+ ✔ SIGNAL_HANDLER,
+ ✖ DEBUG]
```
Comment (iblis17) [inline docs/api/python/libinfo/libinfo.md]: :heart: this

**Comment by ankkhedia** [comment]
@larroy Thanks for your contribution!
@mxnet-label-bot add [pr-awaiting-review, Doc]

File: docs/api/python/libinfo/libinfo.md (hunk 1 above)
Comment (larroy) [reply docs/api/python/libinfo/libinfo.md]: I copied from you :-)

File: python/mxnet/runtime.py [hunk 2]
```diff
@@ -28,21 +29,48 @@ class LibFeature(ctypes.Structure):
...
+    :param x: str The name of a valid feature as string for example 'CUDA'
+
+    Returns
+    -------
+    :return: bool True if it's enabled, False if it's disabled, RuntimeError if the feature is not known
+    """
+    feature_dict = {f.name: f.enabled for f in libinfo_features()}
+    if tocheck not in feature_dict:
+        raise RuntimeError("Feature '{}' is unknown, known features are: {}".format(tocheck, list(feature_dict.keys())))
+    return feature_dict[tocheck]
```
Comment (szha) [inline python/mxnet/runtime.py]: if a dictionary is easier to operate, should the libinfo_features return a dictionary?

Comment (larroy) [reply python/mxnet/runtime.py]: I prefer to keep ordering as in a list. Going from list to dict is oneliner.

Comment (szha) [reply python/mxnet/runtime.py]: is the order information useful? I always thought of it as a set. also, by defining your own data structure class, you can define more pythonic behaviors such as `__contains__` and `__getitem__` to make it easier for feature check.

Comment (larroy) [reply python/mxnet/runtime.py]: They are grouped if you see the enum definition as some features are related, semantically is a set as you describe. For using __contains__ as you propose we would have to wrap the list in a containing class. I don't see a lot of value on complicating this further TBH. Could you provide an example of what you have in mind if you feel strongly about it?

Comment (szha) [reply python/mxnet/runtime.py]: I don't feel stongly about the change. What I had in mind was something like:
```python
>>> features = mx.runtime.libinfo_features()
>>> 'NCCL' in features
True
>>> 'nccl' in features
True
>>> nccl_idx = 2;
>>> nccl_idx in features
True
>>> features['NCCL']
✔ NCCL
>>> features['NCCL'].index
2
>>> features['NCCL'].enabled
True
```

Comment (larroy) [reply python/mxnet/runtime.py]: I see, that looks nice. On the upside repeated calls are more efficient than my proposal. On the downside I think the primary case which is to check if a feature is enabled is less ergonomic due to the syntactic sugar. I think a simple function with a single arg is always clearer.  A good point is that it should be case insensitive. I'm a bit torn.

Comment (apeforest) [reply python/mxnet/runtime.py]: Just curious is the index information useful to user? Why do we even need to expose index?

File: tests/python/unittest/test_runtime.py [hunk 3]
```diff
@@ -29,6 +29,17 @@ def test_libinfo_features():
     ok_(type(features) is list)
     ok_(len(features) > 0)
 
+def test_is_enabled():
```
Comment (apeforest) [inline tests/python/unittest/test_runtime.py]: Can this test run in a docker image with CUDA on the machine without GPU?

File: python/mxnet/runtime.py (hunk 2 above)
Comment (larroy) [reply python/mxnet/runtime.py]: @apeforest I just removed it.

File: tests/python/unittest/test_runtime.py (hunk 3 above)
Comment (larroy) [reply tests/python/unittest/test_runtime.py]: Not sure I get what's the intention.

**Comment by larroy** [comment]
I made the suggested changed, please have another look and merge if you are good with it, thanks.

File: docs/api/python/libinfo/libinfo.md [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+
+In [6]:
+```
+
+
+```eval_rst
+.. autosummary::
+    :nosignatures:
+
+    LibFeature
```
Comment (szha) [inline docs/api/python/libinfo/libinfo.md]: outdated.

**Comment by larroy** [comment]
Thanks for the merge and the review.

**Pull request merged**